File: backtracking/N&M7.cpp
#include<bits/stdc++.h>
using namespace std;

int n,m;
int num[10];
int arr[10];

void solve(int k){
    if(k==m){
        for(int i=0;i<m;i++) cout<<num[arr[i]]<<" ";
        cout<<"\n";
        return;
    }

    for(int i=0;i<n;i++) {
        arr[k] = i;
        solve(k+1);
    }
}

int main(void){
    cin>>n>>m;
    for(int i=0;i<n;i++) cin>>num[i];
    sort(num, num+n);
    solve(0);
}

File: backtracking/N&M7.py
import sys
input = sys.stdin.readline

def solve(k,n,m,arr,num):
    if k==m:
        for i in range(m):
            print(num[arr[i]], end=" ")
        print()
        return
    
    for i in range(n):
        arr[k] = i
        solve(k+1,n,m,arr,num)

def main():
    n,m = map(int, input().split())
    num = list(map(int, input().split()))
    num = sorted(num)
    arr = [0]*n
    solve(0,n,m,arr,num)

if __name__ == "__main__":
    main()
